pysindy/pysindy.py

```diff
@@ -91,18 +91,21 @@ class SINDy(BaseEstimator):
     0.999999985520653
     """
 
-    def __init__(
-        self,
-        optimizer=STLSQ(),
-        feature_library=PolynomialFeatures(),
-        differentiation_method=FiniteDifference(),
-        feature_names=None,
-        discrete_time=False,
-        n_jobs=1,
-    ):
-        self.optimizer = optimizer
-        self.feature_library = feature_library
-        self.differentiation_method = differentiation_method
+    def __init__(self, optimizer=None, feature_library=None,
+                 differentiation_method=None, feature_names=None,
+                 discrete_time=False, n_jobs=1):
+        if optimizer is None:
+            self.optimizer = STLSQ()
+        else:
+            self.optimizer = optimizer
```

Comment:
I think using the ternary operator improves readability here:

```suggestion
        self.optimizer = optimizer if optimizer is not None else STLSQ()
```

Reply:
That works.  Or like this:
```python
if optimizer is None:
    optimizer = STLSQ()
self.optimizer = optimizer
```

Reply:
I like the idea of trying to make this part of the code as short and readable as possible since the `SINDy` class is the most used object in the package. I think the best solution is the second one suggested by @billtubbs. I like @Ohjeah's implementation, but if we use it for the `feature_library` and `differentiation_method` attributes, we need to split up the code into multiple lines anyway to keep the line length under 88. I'll add my own suggested implementation.